moderation/commands.go

```diff
@@ -925,7 +933,17 @@ func AdvancedDeleteMessages(channelID int64, filterUser int64, regex string, toI
 			continue
 		}
 
-		toDelete = append(toDelete, msgs[i].ID)
+		// If attachments only filter enabled, add only messages with attachments, else behave as usual
+		if attachmentFilterEnable {
+			if len(msgs[i].Attachments) > 0 {
+				toDelete = append(toDelete, msgs[i].ID)
+			} else {
+				continue
+			}
+		} else {
+                	toDelete = append(toDelete, msgs[i].ID)
+		}
+
```

Comment:
Why like that? Could easily stick to the current style by using the power of `continue`.

```go
// Check if attachment to ignore
if attachmentFilterEnable && len(msgs[i].Attachments) == 0 {
	continue
}
```

Reply:
@DevKurka

> 
> 
> Why like that? Could easily stick to the current style by using the power of `continue`.
> 
> ```go
> // Check if attachment to ignore
> if attachmentFilterEnable && len(msgs[i].Attachments) == 0 {
> 	continue
> }
> ```

You're 100% correct, I've amended it after testing.